File: Java/Police.java
public class Police extends Person {
    private String badgeNumber;
    private String rank;

    // Default constructor
    public Police() {
        super(); // memanggil constructor Person()
        this.badgeNumber = "";
        this.rank = "";
    }

    // Constructor dengan parameter (include parent)
    public Police(int idPerson, String name, String address, String badgeNumber, String rank) {
        super(idPerson, name, address); // panggil constructor Person
        this.badgeNumber = badgeNumber;
        this.rank = rank;
    }

    // Setter
    public void setBadgeNumber(String badgeNumber) {
        this.badgeNumber = badgeNumber;
    }

    public void setRank(String rank) {
        this.rank = rank;
    }

    // Getter
    public String getBadgeNumber() {
        return this.badgeNumber;
    }

    public String getRank() {
        return this.rank;
    }

    // Display
    public void displayPolice(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "Police ID  : " + getIdPerson());
        System.out.println(sp + "Name       : " + getName());
        System.out.println(sp + "Address    : " + getAddress());
        System.out.println(sp + "Badge No   : " + getBadgeNumber());
        System.out.println(sp + "Rank       : " + getRank());
        System.out.println();
    }
}

File: Python/Police.py
from Person import Person

class Police(Person):
    def __init__(self, id_person=0, name="", address="", badge_number="", rank=""):
        # Panggil constructor parent (Person)
        super().__init__(id_person, name, address)
        self.badge_number = badge_number
        self.rank = rank

    # Setter
    def set_badge_number(self, badge_number):
        self.badge_number = badge_number

    def set_rank(self, rank):
        self.rank = rank

    # Getter
    def get_badge_number(self):
        return self.badge_number

    def get_rank(self):
        return self.rank

    # Display
    def display_police(self, indent=0):
        sp = " " * indent
        print(f"{sp}Police ID  : {self.get_id_person()}")
        print(f"{sp}Name       : {self.get_name()}")
        print(f"{sp}Address    : {self.get_address()}")
        print(f"{sp}Badge No   : {self.get_badge_number()}")
        print(f"{sp}Rank       : {self.get_rank()}\n")

    def __del__(self):
        # Destructor opsional di Python
        pass
